src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -57,19 +65,65 @@ seastar::future<> ClientRequest::start()
       return with_blocking_future(osd.wait_for_pg(m->get_spg()));
     }).then([this, ref=std::move(ref)](Ref<PG> pg) {
       return seastar::do_with(
-	std::move(pg), std::move(ref), [this](auto pg, auto op) {
-	return with_blocking_future(
-	  handle.enter(pp(*pg).await_map)
-	).then([this, pg] {
+	std::move(pg), std::move(ref), [this](auto pgref, auto opref) {
```

Comment:
I'm passing it into the do_with in order to ensure that both it and the pg ref survive until after the last future.

I'll rename the earlier variables so that they'll match.